tests/src/test/java/io/streamnative/pulsar/handlers/kop/docker/PulsarContainer.java

```diff
@@ -140,18 +140,21 @@ public void start() throws Exception {
 
         // for KOP broker to broker communications via TLS
         pulsarContainer.withEnv("PULSAR_PREFIX_kopSslTruststoreLocation", "/pulsar/conf/ca.jks");
-        pulsarContainer.withEnv("PULSAR_PREFIX_kopSslTruststorePassword", "");
+        pulsarContainer.withEnv("PULSAR_PREFIX_kopSslTruststorePassword", "pulsar");
 
         pulsarContainer.withEnv("PULSAR_PREFIX_brokerServiceURLTLS", "pulsar+ssl://pulsar:6651");
         pulsarContainer.withEnv("PULSAR_PREFIX_brokerWebServiceURLTLS", "https://pulsar:8443");
         pulsarContainer.withEnv("PULSAR_PREFIX_brokerServicePortTls", "6651");
         pulsarContainer.withEnv("PULSAR_PREFIX_webServicePortTls", "8443");
 
-        pulsarContainer.withEnv("PULSAR_PREFIX_tlsAllowInsecureConnection", "true");
-        pulsarContainer.withEnv("PULSAR_PREFIX_tlsHostnameVerificationEnabled", "false");
+        pulsarContainer.withEnv("PULSAR_PREFIX_tlsAllowInsecureConnection", "false");
+        pulsarContainer.withEnv("PULSAR_PREFIX_tlsHostnameVerificationEnabled", "true");
 
         pulsarContainer.withEnv("PULSAR_PREFIX_kopTlsEnabledWithBroker", "true");
         pulsarContainer.withEnv("PULSAR_PREFIX_tlsEnabledWithBroker", "true");
+        pulsarContainer.withEnv("PULSAR_PREFIX_brokerClientTlsEnabledWithKeyStore", "true");
+        pulsarContainer.withEnv("PULSAR_PREFIX_brokerClientTlsTrustStore", "/pulsar/conf/ca.jks");
+        pulsarContainer.withEnv("PULSAR_PREFIX_brokerClientTlsTrustStorePassword", "pulsar");
```

Comment:
@eolivelli - the test passes with and without this config. I suspect it is because we do not actually have broker to broker networking in this test. I think that is likely fine, but I just want to mention it.

Reply:
in Pulsar there is mostly always a broker-to-broker communication when you start the broker and it creates the default entities (tenants, namespaces).
In fact usually when you enable authentication and you misconfigure it the broker is not able to start.

Maybe we are still enabling the non-TLS endpoint and by default we prefer to go with the plain socket connection ?

Reply:
That could be the case.